static/js/composebox_typeahead.js

```diff
@@ -358,16 +360,16 @@ exports.content_typeahead_selected = function (item) {
     if (this.completing === 'emoji') {
         // leading and trailing spaces are required for emoji, except if it begins a message.
         if (beginning.lastIndexOf(":") === 0 || beginning.charAt(beginning.lastIndexOf(":") - 1) === " ") {
-            beginning = beginning.replace(/:\S+$/, "") + ":" + item.emoji_name + ": ";
+            beginning = (beginning.substring(0, beginning.length - this.token.length - 1)+ ":" + item.emoji_name + ": ");
```

Comment:
The regex gets complex when we are trying to capture emoji names with multiple spaces. 
So using the substring method is better. Also, it's similar to how other auto completions such *mention* and *stream* are captured.